includes/functions/llms.functions.access.php

```diff
@@ -321,27 +324,38 @@ function llms_is_post_restricted_by_drip_settings( $post_id, $user_id = null ) {
 
 	$post_type = get_post_type( $post_id );
 
-	// if we're on a lesson, lesson id is the post id.
-	if ( 'lesson' === $post_type ) {
-		$lesson_id = $post_id;
-	} elseif ( 'llms_quiz' === $post_type ) {
-		$quiz      = llms_get_post( $post_id );
-		$lesson_id = $quiz->get( 'lesson_id' );
-		if ( ! $lesson_id ) {
+	switch ( $post_type ) {
+		// If we're on a lesson, lesson id is the post id.
+		case 'lesson':
+			$lesson_id = $post_id;
+			break;
+		case 'llms_quiz':
+			$quiz      = llms_get_post( $post_id );
+			$lesson_id = $quiz->get( 'lesson_id' );
+			if ( ! $lesson_id ) {
+				return false;
+			}
+			break;
+		default: // Don't pass other post types.
 			return false;
-		}
-	} else {
-		// dont pass other post types in here dumb dumb.
-		return false;
 	}
 
-	$lesson = new LLMS_Lesson( $lesson_id );
+	$lesson  = new LLMS_Lesson( $lesson_id );
+	$user_id = $user_id ?? get_current_user_id();
+	/**
+	 * Filters whether or not to bypass drip restrictions on completed lessons.
+	 *
+	 * @since [version]
+	 *
+	 * @param boolean $drip_bypass Whether or not to bypass drip restrictions on completed lessons.
+	 * @param int     $post_id     WP Post ID of a lesson or quiz potentially restricted by drip settings.
+	 * @param int     $user_id     WP User ID.
+	 */
+	$is_available = apply_filters( 'llms_lesson_drip_bypass_if_completed', true, $post_id, $user_id ) &&
+	                llms_is_complete( $user_id, $lesson_id, 'lesson' ) ||
+	                $lesson->is_available();
```

Comment:
```suggestion
	/**
	 * Filters whether or not to bypass drip restrictions on completed lessons.
	 *
	 * @since [version]
	 *
	 * @param boolean $drip_bypass Whether or not to bypass drip restrictions on completed lessons.
	 * @param int     $post_id     WP Post ID of a lesson or quiz potentially restricted by drip settings.
	 * @param int     $user_id     WP User ID.
	 */
	$is_available = apply_filters( 'llms_lesson_drip_bypass_if_completed', true, $post_id, $user_id );
	
	$is_available = ( $user_id && llms_is_complete( $user_id, $lesson_id, 'lesson' ) ) || $lesson->is_available();
```

Make the previous line more readable and don't run the `is_complete()` check against a logged out user. This would happen if there's not user id passed into the function and there's no logged in user.